internal/controller/metrics.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2025.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	inferencev1alpha1 "github.com/openshift/instaslice-operator/api/v1alpha1"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// updateMetrics - updates UpdateDeployedPodTotalMetrics, UpdateGpuSliceMetrics and UpdateCompatibleProfilesMetrics
+func (r *InstasliceReconciler) updateMetrics(ctx context.Context, instasliceList inferencev1alpha1.InstasliceList) error {
+	log := logr.FromContext(ctx)
+	for _, instaslice := range instasliceList.Items {
```

Comment:
Why are we updating all InstaSlice objects in the cluster?

Reply:
We are not updating the instaslice object, we are just fetching the latest state before updating Prometheus metrics and the reason is, UpdateCompatibleProfilesMetrics metrics store metrics per node and to ensure Metrics Are Always Up-to-Date.